app/controllers/events_controller.rb

```diff
@@ -101,6 +101,9 @@ def ranking
   end
 
   def overview
+    if @event.type == 'League'
```

Comment:
@birneamstiel I like the approach that @jonasnoki followed in #460 – he created an explicit subclass of EventsController in order to control more precisely which requests should be routed to which controller actions (**and especially which should not**). Maybe you could take a look into his changes for inspiration, especially the following change in the routing:

```diff
-  get '/events/:id/ranking', to: 'events#ranking', as: 'event_ranking'
+  get '/leagues/:id/ranking', to: 'leagues#ranking', as: 'leagues_ranking'
```

This would make the redirection you implemented obsolete since the route would simply not exist. However, I also like your approach of redirecting the user instead of confronting him with a 404 page. Dunno … 🙃

Reply:
Hmm I'm not so sure about that... This seems to be a nice way of doing it but it's some serious change in structure for this small thing. I'd also have to create both `RankingListController` and `TournamentController`, correct? Or would you just move the current redirection logic into the `LeagueController`?

Reply:
Yeah, you're right – let's just stick with your implementation. 🙂